controllers/admin/AdminCartRulesController.php

```diff
@@ -640,9 +640,10 @@ public function renderForm()
         $groups = $current_object->getAssociatedRestrictions('group', false, true);
         $shops = $current_object->getAssociatedRestrictions('shop', false, false);
         $cart_rules = $current_object->getAssociatedRestrictions('cart_rule', false, true, 0, $limit);
-        $carriers = $current_object->getAssociatedRestrictions('carrier', true, false);
+        $carriers = $current_object->getAssociatedRestrictions('carrier', true, true);
         foreach ($carriers as &$carriers2) {
             foreach ($carriers2 as &$carrier) {
+                $carrier['name'] = $carrier['id_carrier'].' - '.$carrier['name'].' ('.$carrier['delay'].')';
```

Comment:
This line breaks the test made two lines below, it must be moved in the foreach in order to change the value after `PS_SHOP_NAME` is applied

Reply:
Fixed!